### ISSUE TITLE
auto_pause_delay_in_minutes property has value of 0 when disabled, not -1

### ISSUE THREAD

**Issue by rit-bit** [hashicorp/terraform-provider-azurerm#27241]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.5

### AzureRM Provider Version

3.108.0

### Affected Resource(s)/Data Source(s)

azurerm_mssql_database

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "this" {
  name     = "example-rg"
  location = "UK South"
}

resource "azurerm_mssql_server" "example" {
  name                          = "example-server"
  resource_group_name           = azurerm_resource_group.this.name
  location                      = "UK South"
  version                       = "12.0"
  administrator_login           = var.administrator_login
  administrator_login_password  = var.administrator_login_password
  public_network_access_enabled = true
  minimum_tls_version           = "1.2"

  lifecycle {
    prevent_destroy = true
  }
}


resource "azurerm_mssql_database" "example" {
  name         = "example-db"
  server_id    = azurerm_mssql_server.example.id
  collation    = "SQL_Latin1_General_CP1_CI_AS"
  license_type = "LicenseIncluded"
  sku_name             = var.database_sku
  storage_account_type = var.database_backup_storage_redundancy

  auto_pause_delay_in_minutes = -1

  short_term_retention_policy {
    retention_days = 7
  }

  long_term_retention_policy {
    weekly_retention  = "P2M"
    monthly_retention = "P6M"
    yearly_retention  = "P2Y"
    week_of_year      = 1
  }

  lifecycle {
    prevent_destroy = true
  }
}
```

### Debug Output/Panic Output

```shell
n/a
(I'm happy to provide this if genuinely helpful but it seems that the problem is fairly apparent without the need for this)
```

### Expected Behaviour

Re-running `terraform plan` should result in no changes being detected:
```
No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.
```

### Actual Behaviour

Re-running `terraform plan` results in 1 planned change:
```
Terraform will perform the following actions:

  # module.database.azurerm_mssql_database.database will be updated in-place
  ~ resource "azurerm_mssql_database" "database" {
      ~ auto_pause_delay_in_minutes                                = 0 -> -1
      ...
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

If I run `terraform apply`, terraform does seem to apply this change (taking 60s+) and it doesn't error. But running `terraform plan` again immediately afterwards still shows the above change.

---

If I instead specify `auto_pause_delay_in_minutes` as `0`, then terraform correctly sees that there are no changes to make, but if I change the database SKU and run `terraform apply`, I get this error:
```
Enabling auto-pause for a serverless database is not supported if long-term backup retention is enabled.
```

### Steps to Reproduce

1. Run `terraform apply` to create the example resource group, server, and database
2. Run `terraform apply` again

**Comment by sinbai** [comment]
Hi @rit-bit thanks for raising this issue. First, I would like to clarify that the `auto_pause_delay_in_minutes` property is only settable for Serverless databases. [Here](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mssql_database#auto_pause_delay_in_minutes) is the description document for `auto_pause_delay_in_minutes`. Could you please confirm if it is set for Serverless databases? Also, I confirmed that when I disabled `auto-pause`, the value of `autoPauseDelay` is `-1` in Azure Portal. Could you please double check? Or could you please provide explanation why you think it should be `0` instead of `-1`?
![image](https://github.com/user-attachments/assets/54f8d8a9-1296-402f-85d4-9f25adbb2ad9)
![image](https://github.com/user-attachments/assets/e9a0ece1-d1b1-4585-a39a-23681d89aed8)

**Comment by sinbai** [comment]
Hi @rit-bit thanks for your explanation. I would like to explain that the value of `auto_pause_delay_in_minutes` is set to `0` in state file because it is an int variable, and the default of int is 0 even though it is not specified in the config. Also, is it possible to provide the complete tf (including variable values) configs and repro steps to help reproduce the following error and troubleshoot?

`Enabling auto-pause for a serverless database is not supported if long-term backup retention is enabled.`

**Comment by sinbai** [comment]
Hi @rit-bit thanks for your reply. I reproduced the mentioned error. 

First, I would like to explain that Terraform manages Azure resource through the Azure Rest API. However, per the Terraform log below, we can see that the error is returned by Azure Rest API, not Terraform，even though Terraform didn't send any value for `auto_pause_delay_in_minutes` in step 3.

- Request from Terraform in step 3:

`PATCH /subscriptions/.../resourceGroups/example-rg-2741/providers/Microsoft.Sql/servers/example-server-27241/databases/example-example27241?api-version=2023-02-01-preview HTTP/1.1`

`{"properties":{},"sku":{"name":"HS_S_Gen5_2"}}: timestamp="2024-09-03T14:15:05.415+0800"`

- Response from Azure Rest API in step 3:

`{"name":"618d6869-cfa5-4126-bca0-e0ed016dca4f","status":"Failed","startTime":"2024-09-03T06:15:08.257Z","error":{"code":"UpdateToServerlessIfLtrIsNotDisabled","message":"Enabling auto-pause for a serverless database is not supported if long-term backup retention is enabled."}}: timestamp="2024-09-03T14:15:23.803+0800"`

For the error in step 3, as set in step 4 above, we can solve this issue by disabling `auto_pause_delay_in_minutes`.

For the symptoms in step 5 above, this is because the Azure Rest API doesn't return the value of `auto_pause_delay_in_minutes` in step 4. For this case, on Terraform side, we recomment setting `ignore_changes` as shown below to avoid this situation.

 ```
 lifecycle {
    ignore_changes = [auto_pause_delay_in_minutes]
  }
```

Thank you again for raising this issue.

**Comment by rit-bit** [comment]
Hi @sinbai

I'm pleased you were able to reproduce the error.

Regarding your suggestion to use the `ignore_changes` lifecycle property to prevent the symptom from step 5: this is something I have tried and it does work for cases where you are not trying to change anything on the database - terraform correctly identifies that it shouldn't change any properties on the database resource.

If however I am using the `ignore_changes` lifecycle property as above and I want to change the SKU for example from "HS_S_Gen5_2" to "HS_S_Gen5_4", then terraform encounters the same error as before:
```
Enabling auto-pause for a serverless database is not supported if long-term backup retention is enabled.
```
If I remove the `ignore_changes` lifecycle property (and keep the line `auto_pause_delay_in_minutes = -1`) then the apply succeeds, but I have to re-add the `ignore_changes` lifecycle property afterwards to prevent the symptom from step 5 surfacing again.

It's undesirable and very impractical to have to add or remove the `ignore_changes` lifecycle property whenever a change is to be made to the database resource.

**Comment by sinbai** [comment]
Yes, it is indeed what you said. Another way is to manually modify the `auto_pause_delay_in_minutes` in the state file to `-1` after step 4, or submit an issue in the[ Azure Rest API repo](https://github.com/Azure/azure-rest-api-specs/issues), requesting the API to return the specified value in the response if `auto_pause_delay_in_minutes` is specified in the API request. Apart from this, as far as I know, there is no better solution.

**Comment by JonathonAnderson** [comment]
@sinbai it seems like there should be a solution here where Terraform is able to recognize that the value is missing from the API response, examine the Terraform configs, and then update the state file accordingly. All of the information is available, so it's not a matter of ability.
